packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -587,9 +587,41 @@ export const isDarkRgbColour = string_rgb => {
 };
 /* eslint-enable */
 
-export const removeExtraInput = instance => {
+export const appendCollapsedMainBlocksFields = block_instance => {
+    try {
+        // Return if the block is not collapsed
+        if (!block_instance?.collapsed_) return;
+        const type_of_block = block_instance?.getField(block_instance.type);
+        if (type_of_block) return;
+
+        const [block_image, block_name] = block_instance?.inputList[0]?.fieldRow.map(field => field.value_);
+        const collapsed_field = block_instance?.getField(Blockly.constants.COLLAPSED_FIELD_NAME);
+        const collapsed_input = block_instance?.getInput(Blockly.constants.COLLAPSED_INPUT_NAME);
+
+        // Initialize the icon_added property if not already done
+        if (collapsed_input && collapsed_field) {
+            // Add the image and label fields
+            collapsed_input.appendField(new Blockly.FieldImage(block_image, 25, 25, '', ''));
+            collapsed_input.appendField(new Blockly.FieldLabel(block_name, 'blocklyTextRootBlockHeaderCollapsed'));
+
+            // Add the dropdown icon
+            const dropdown_path = `${block_instance?.workspace.options.pathToMedia}dropdown-arrow.svg`;
+            const field_expand_icon = new Blockly.FieldImage(dropdown_path, 16, 16, localize('Collapsed'), () =>
+                block_instance?.setCollapsed(false)
+            );
+            collapsed_input.appendField(field_expand_icon, block_instance.type);
+            collapsed_input.sourceBlock.width = 300;
+            // hide the default collapsed field generated by Blockly
+            collapsed_field.setVisible(false);
+        }
+    } catch (e) {
+        // eslint-disable-next-line
+        console.log(e);
+    }
```

Comment:
@rupato-deriv There is a console.log(e) in the catch.

Reply:
done ✅